File: 118.杨辉三角.py
#
# @lc app=leetcode.cn id=118 lang=python3
#
# [118] 杨辉三角
#

# @lc code=start
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = list()
        for i in range(numRows):
            row = list()
            for j in range(0, i + 1):
                if j == 0 or j == i:    # 边界直接加入 1
                    row.append(1)
                else:   # 其余情况正常规则计算
                    row.append(result[i-1][j] + result[i-1][j-1])
            result.append(row)
        return result
# @lc code=end

File: 118.杨辉三角.go
/*
 * @lc app=leetcode.cn id=118 lang=golang
 *
 * [118] 杨辉三角
 */

// @lc code=start
func generate(numRows int) (res [][]int) {
	for i := 0; i < numRows; i++ {
		row := []int{} // 初始化本行
		for j := 0; j < i+1; j++ {
			if j == 0 || j == i { // 边界则加入 1
				row = append(row, 1)
			} else { // 其余情况按照要求加入左上方与右上方之和
				row = append(row, res[i-1][j]+res[i-1][j-1])
			}
		}
		res = append(res, row)
	}
	return
}

// @lc code=end

